File: client/web/sudoo/lib/domain/model/discovery/supplier.dart
import 'package:sudoo/domain/model/discovery/product.dart';
import 'package:sudoo/domain/model/user/location.dart';

class Supplier {
  final String supplierId;
  final String name;
  final String avatar;
  final String brand;
  final Location? location;
  final String contactUrl;
  final String locationName;
  final int? totalProducts;
  final double? rate;
  final List<Product>? products;

  const Supplier(
    this.supplierId,
    this.name,
    this.avatar,
    this.brand,
    this.location,
    this.contactUrl,
    this.locationName,
    this.totalProducts,
    this.rate, {
    this.products,
  });
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/discovery/Supplier.kt
package com.sudo248.sudoo.domain.entity.discovery

import com.sudo248.sudoo.domain.entity.user.Address
import java.time.LocalDateTime

data class Supplier(
    val supplierId: String,
    val name: String,
    val avatar: String,
    val brand: String,
    val contactUrl: String,
    val totalProducts: Int,
    val rate: Float,
    val createAt: LocalDateTime,
    val address: Address
)